### PR TITLE
`FEATURE` Introduced the child parameters to the radial gauge.

### PR DESCRIPTION
**Pull request by kamilklyta**
### Changes:
- Allow `RadialGauge` to pass the child as a label.
  - Added `child` parameter to the `RadialGauge` widget.
  - Removed `style` and `labelProvider` arguments.
  - Migrated `GaugeLabelProvider` class.
  - Added `child` and `builder` parameters to the `AnimatedRadialGauge` widget.
  - Introduced `RadialGaugeLabel` widget.
- The `AnimatedRadialGauge` value is now constrained by the `min` and `max` parameters.

### CODE REVIEW

File: lib/src/radial_gauge/data/gauge_label_provider.dart [hunk 1]
```diff
@@ -1,40 +1,39 @@
 import 'package:collection/collection.dart';
 import 'package:flutter/material.dart';
 
+/// Provides string labels for the `RadialGauge`.
 abstract class GaugeLabelProvider {
   const GaugeLabelProvider();
 
   const factory GaugeLabelProvider.value({
-    Curve curve,
     int fractionDigits,
```
Comment (KopaczP) [inline lib/src/radial_gauge/data/gauge_label_provider.dart:9]: decimalPlaces?

File: lib/src/radial_gauge/data/gauge_label_provider.dart [hunk 2]
```diff
@@ -1,40 +1,39 @@
 import 'package:collection/collection.dart';
 import 'package:flutter/material.dart';
 
+/// Provides string labels for the `RadialGauge`.
 abstract class GaugeLabelProvider {
```
Comment (KopaczP) [inline lib/src/radial_gauge/data/gauge_label_provider.dart:5]: While I like the idea, I dont think that hardcoding decimal places is a good idea. We take out some of the freedom of implementation.

File: lib/src/radial_gauge/data/gauge_label_provider.dart (hunk 1 above)
Comment (kamilklyta) [reply lib/src/radial_gauge/data/gauge_label_provider.dart:9]: This is consistent with a `toStringAsFixed` function.


Screen Shot 2022-06-03 at 09 09 06

File: lib/src/radial_gauge/data/gauge_label_provider.dart (hunk 2 above)
Comment (KopaczP) [reply lib/src/radial_gauge/data/gauge_label_provider.dart:5]: Something unsticked in my mind.

**Review by KopaczP — APPROVED** [review]

**Review by KopaczP — APPROVED** [review]

**Pull request merged**